### ISSUE TITLE
"An error occured" is shown every time users see a Pull Request

### ISSUE THREAD

**Issue by samueljmarshall** [tomasbjerre/simple-stash-commit-checker#41]
Users are getting "An error occured" dialog every time they go to view a pull request.

After Developers click OK on the error box, the Pull Requests still displays as expected and can be approved and merged.

However the same error *may* be shown on merge even though the merge still works. This happens to all our users who are not project admins.

The stack trace below is shown in the **atlassian-stash.log** every time the pop up is displayed on Stash UI:
``` java
2015-07-22 09:26:13,723 ERROR [http-nio-7990-exec-4] username @17OSY73x566x32904x1 1jnhn0k 192.168.12.173,127.0.0.1 "GET /rest/api/latest/projects/PROJ/repos/gradle-common-settings/pull-requests/2/merge HTTP/1.1" s.b.s.SsccRepositoryMergeRequestCheck
com.atlassian.stash.exception.AuthorisationException: You are not permitted to access this resource
        at com.atlassian.stash.internal.aop.ExceptionRewriteAdvice.afterThrowing(ExceptionRewriteAdvice.java:36) ~[stash-platform-3.10.2.jar:na]
        at se.bjurr.sscc.SsccRepositoryMergeRequestCheck.check(SsccRepositoryMergeRequestCheck.java:53) ~[na:na]
        at com.atlassian.stash.internal.pull.DefaultMergeRequestCheckService.doCheck(DefaultMergeRequestCheckService.java:73) [stash-service-impl-3.10.2.jar:na]
        at com.atlassian.stash.internal.pull.DefaultMergeRequestCheckService.checkMergeability(DefaultMergeRequestCheckService.java:57) [stash-service-impl-3.10.2.jar:na]
        at com.atlassian.stash.internal.pull.DefaultPullRequestService.canMerge(DefaultPullRequestService.java:365) [stash-service-impl-3.10.2.jar:na]
        at com.atlassian.stash.internal.rest.pull.PullRequestResource.canMerge(PullRequestResource.java:431) [stash-rest-3.10.2.jar:na]
        at com.atlassian.applinks.core.rest.context.ContextFilter.doFilter(ContextFilter.java:25) [applinks-plugin-4.3.7_1435097926000.jar:na]
        at com.atlassian.stash.internal.spring.security.StashAuthenticationFilter.doFilter(StashAuthenticationFilter.java:88) [StashAuthenticationFilter.class:na]
        at com.atlassian.stash.internal.web.auth.BeforeLoginPluginAuthenticationFilter.doInsideSpringSecurityChain(BeforeLoginPluginAuthenticationFilter.java:111) [BeforeLoginPluginAuthenticationFilter.class:na]
        at com.atlassian.stash.internal.web.auth.BeforeLoginPluginAuthenticationFilter.doFilter(BeforeLoginPluginAuthenticationFilter.java:77) [BeforeLoginPluginAuthenticationFilter.class:na]
        at com.atlassian.security.auth.trustedapps.filter.TrustedApplicationsFilter.doFilter(TrustedApplicationsFilter.java:103) [atlassian-trusted-apps-core-4.0.0.jar:na]
        at com.atlassian.oauth.serviceprovider.internal.servlet.OAuthFilter.doFilter(OAuthFilter.java:79) [atlassian-oauth-service-provider-plugin-1.9.10_1435097926000.jar:na]
        at com.atlassian.analytics.client.filter.DefaultAnalyticsFilter.doFilter(DefaultAnalyticsFilter.java:32) [analytics-client-3.65.1_1435097928000.jar:na]
        at com.atlassian.analytics.client.filter.AbstractHttpFilter.doFilter(AbstractHttpFilter.java:32) [analytics-client-3.65.1_1435097928000.jar:na]
        at com.atlassian.stash.internal.web.StreamGuardFilter.doFilter(StreamGuardFilter.java:49) [stash-stream-guard-plugin-1.0.1.jar:na]
        at com.atlassian.stash.internal.web.auth.BeforeLoginPluginAuthenticationFilter.doBeforeBeforeLoginFilters(BeforeLoginPluginAuthenticationFilter.java:89) [BeforeLoginPluginAuthenticationFilter.class:na]
        at com.atlassian.stash.internal.web.auth.BeforeLoginPluginAuthenticationFilter.doFilter(BeforeLoginPluginAuthenticationFilter.java:75) [BeforeLoginPluginAuthenticationFilter.class:na]
        at com.atlassian.stash.internal.request.DefaultRequestManager.doAsRequest(DefaultRequestManager.java:85) [stash-service-impl-3.10.2.jar:na]
        at com.atlassian.stash.internal.hazelcast.ConfigurableWebFilter.doFilter(ConfigurableWebFilter.java:38) [ConfigurableWebFilter.class:na]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_31]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_31]
        ... 304 frames trimmed
Caused by: org.springframework.security.access.AccessDeniedException: Access is denied
        at org.springframework.security.access.vote.AffirmativeBased.decide(AffirmativeBased.java:83) ~[spring-security-core-3.2.7.RELEASE.jar:3.2.7.RELEASE]
        ... 21 common frames omitted
2015-07-22 09:26:13,741 ERROR [http-nio-7990-exec-4] username @17OSY73x566x32904x1 1jnhn0k 192.168.12.173,127.0.0.1 "GET /rest/api/latest/projects/PROJ/repos/gradle-common-settings/pull-requests/2/merge HTTP/1.1" c.a.s.r.e.UnhandledExceptionMapper Unhandled exception while processing REST request: "GET /rest/api/latest/projects/PROJ/repos/gradle-common-settings/pull-requests/2/merge HTTP/1.1"
org.springframework.transaction.UnexpectedRollbackException: Transaction rolled back because it has been marked as rollback-only
        at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:720) ~[spring-tx-4.1.6.RELEASE.jar:4.1.6.RELEASE]
        at com.atlassian.stash.internal.rest.pull.PullRequestResource.canMerge(PullRequestResource.java:431) ~[stash-rest-3.10.2.jar:na]
        at com.atlassian.applinks.core.rest.context.ContextFilter.doFilter(ContextFilter.java:25) [applinks-plugin-4.3.7_1435097926000.jar:na]
        at com.atlassian.stash.internal.spring.security.StashAuthenticationFilter.doFilter(StashAuthenticationFilter.java:88) [StashAuthenticationFilter.class:na]
        at com.atlassian.stash.internal.web.auth.BeforeLoginPluginAuthenticationFilter.doInsideSpringSecurityChain(BeforeLoginPluginAuthenticationFilter.java:111) [BeforeLoginPluginAuthenticationFilter.class:na]
        at com.atlassian.stash.internal.web.auth.BeforeLoginPluginAuthenticationFilter.doFilter(BeforeLoginPluginAuthenticationFilter.java:77) [BeforeLoginPluginAuthenticationFilter.class:na]
        at com.atlassian.security.auth.trustedapps.filter.TrustedApplicationsFilter.doFilter(TrustedApplicationsFilter.java:103) [atlassian-trusted-apps-core-4.0.0.jar:na]
        at com.atlassian.oauth.serviceprovider.internal.servlet.OAuthFilter.doFilter(OAuthFilter.java:79) [atlassian-oauth-service-provider-plugin-1.9.10_1435097926000.jar:na]
        at com.atlassian.analytics.client.filter.DefaultAnalyticsFilter.doFilter(DefaultAnalyticsFilter.java:32) [analytics-client-3.65.1_1435097928000.jar:na]
        at com.atlassian.analytics.client.filter.AbstractHttpFilter.doFilter(AbstractHttpFilter.java:32) [analytics-client-3.65.1_1435097928000.jar:na]
        at com.atlassian.stash.internal.web.StreamGuardFilter.doFilter(StreamGuardFilter.java:49) [stash-stream-guard-plugin-1.0.1.jar:na]
        at com.atlassian.stash.internal.web.auth.BeforeLoginPluginAuthenticationFilter.doBeforeBeforeLoginFilters(BeforeLoginPluginAuthenticationFilter.java:89) [BeforeLoginPluginAuthenticationFilter.class:na]
        at com.atlassian.stash.internal.web.auth.BeforeLoginPluginAuthenticationFilter.doFilter(BeforeLoginPluginAuthenticationFilter.java:75) [BeforeLoginPluginAuthenticationFilter.class:na]
        at com.atlassian.stash.internal.request.DefaultRequestManager.doAsRequest(DefaultRequestManager.java:85) [stash-service-impl-3.10.2.jar:na]
        at com.atlassian.stash.internal.hazelcast.ConfigurableWebFilter.doFilter(ConfigurableWebFilter.java:38) [ConfigurableWebFilter.class:na]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_31]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_31]
        ... 242 frames trimmed
```
#### Workaround
Disable the Simple [Simple Stash Commit Checker](https://marketplace.atlassian.com/plugins/se.bjurr.sscc.sscc) add-on.

**Comment by tomasbjerre** [comment]
Or just disable checking commits in pull requests.

I will look at it as soon as I find the time.

**Comment by tomasbjerre** [comment]
Interesting. I will look at it in 2 weeks when I get back from vacation.

**Comment by tomasbjerre** [comment]
This should be fixed in 1.11, can you test it?

**Comment by tomasbjerre** [comment]
Closing it, open again if problem still occurs.
